### DISCUSSION TITLE
Creating this snap

### DISCUSSION THREAD

**Discussion by soumyaDghosh** [soumyaDghosh/tube-converter-snap#2] [General]
Okay so, the thing, the snap to run properly, the pythonpath, has to be `/snap/tube-converter/current/usr/lib`. This I guess needs to be changed from the code. Don't ask python about path, as things are all containerized and sandboxed. I think just chagning the path solve this issue.
@fsobolev @nlogozzo

**Comment by fsobolev** [comment]
Could you create a text file in this repo containing a full path to python `.so`, and install this file together with TC? So then I will update the app to contain something like this (pseudo-code):
```
if (File.Exists("/snap/tube-converter/current/python-snap.txt"))
{
    Python.Runtime.PythonDLL = File.Read("/snap/tube-converter/current/python-snap.txt");
}
```

I could hardcode the path instead of using an additional file, but with this file in your repo you will be able to change path to python lib (when it will be required to change python version in snap) without changing upstream code.

**Reply by soumyaDghosh** [reply]
Do one thing, have a check if `snapcraft` exists while building or not. If the `snapcraft` exists while building, use `/snap/tube-converter/current/usr/lib`. I hope this will work.
There is no need of a .txt file. The path just need to have `/snap/tube-converter/current/usr` in addition, and all will be fine. I think checking for snapcraft will be a better option.

**Reply by soumyaDghosh** [reply]
Even what I can do this, I can change the path while pulling, You don't need to hardcode the path, but I will hardcode it. Just tell me which file I need to change. I will use the `sed` command and rewrite it. That's the best option.

**Reply by fsobolev** [reply]
> Even what I can do this, I can change the path while pulling, You don't need to hardcode the path, but I will hardcode it. Just tell me which file I need to change. I will use the `sed` command and rewrite it. That's the best option.

I hope you don't mind, but I would want to do it a bit differently. Instead of using `sed`, you can use environment variable. So in snap yaml you will do:
```
export TC_PYTHON_PATH=/path/to/python/filename.so
just install-snap
```

and I will handle the rest, going to create a PR for testing.
Take note that the path should be full, including filename, not only directory.

**Reply by soumyaDghosh** [reply]
Sorry, It can't be done. To set the Pythonpath manually, I need to use the `python plugin`, not possible in this case.

**Reply by soumyaDghosh** [reply]
Using `sed` will not affect your code, btw

**Reply by fsobolev** [reply]
> Sorry, It can't be done. To set the Pythonpath manually, I need to use the `python plugin`, not possible in this case.

We don't need to set python path. All we need is to use our *custom* env variable to tell TC where the `.so` file is. We can name it differently if you don't like this name, `TC_PYTHON_SO` or anything else.

> Using sed will not affect your code, btw

Yeah, but it's just I don't want to use hardcoded path in the app code that will be overwritten anyway. Using env var is more flexible and clean.

**Reply by soumyaDghosh** [reply]
Yes, I guess that can be done, we cando it. Just tell me the name of the variable.

**Reply by fsobolev** [reply]
https://github.com/nlogozzo/NickvisionTubeConverter/pull/101
